appsync-lambda-bedrock-async-stream-subscription-cdk/lib/appsync-lambda-bedrock-async-stream-subscription-cdk-stack.ts

```diff
@@ -0,0 +1,199 @@
+import * as cdk from 'aws-cdk-lib';
+import * as appsync from 'aws-cdk-lib/aws-appsync';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as logs from 'aws-cdk-lib/aws-logs';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+import { Construct } from 'constructs';
+import * as path from 'path';
+
+export class AppsyncLambdaBedrockAsyncStreamSubscriptionCdkStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const api = new appsync.GraphqlApi(this, 'Api', {
+      name: 'bedrock-streaming-api',
+      definition: appsync.Definition.fromFile(path.join(__dirname, '..', 'schema.graphql')),
+      authorizationConfig: {
+        defaultAuthorization: {
+          authorizationType: appsync.AuthorizationType.API_KEY,
+        },
+      },
+      logConfig: {
+        fieldLogLevel: appsync.FieldLogLevel.ALL, // Change to ALL to see resolver details
+        excludeVerboseContent: false, // Include verbose content
+        retention: logs.RetentionDays.ONE_WEEK
+      },
+      xrayEnabled: false
+    });
+    
+    
+
+    const invocationHandler = new NodejsFunction(this, 'InvocationHandler', {
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: 'handler',
+      entry: path.join(__dirname, 'lambda/invocation/index.ts'),
+      timeout: cdk.Duration.minutes(15),
+      environment: {
+        APPSYNC_ENDPOINT: api.graphqlUrl,
+        APPSYNC_API_KEY: api.apiKey || '',
+      },
+      logRetention: logs.RetentionDays.ONE_WEEK,
+      tracing: lambda.Tracing.DISABLED,
+      bundling: {
+        minify: true,
+        sourceMap: false // Disable source maps to reduce log size
+      }
+    });
+    
+
+    // Add Bedrock permissions to Lambda
+    invocationHandler.addToRolePolicy(new iam.PolicyStatement({
+      actions: ['bedrock:InvokeModelWithResponseStream'],
+      resources: [
+        'arn:aws:bedrock:us-east-1::foundation-model/anthropic.claude-3-5-sonnet-20241022-v2:0',
+        'arn:aws:bedrock:us-east-2::foundation-model/anthropic.claude-3-5-sonnet-20241022-v2:0',
+        'arn:aws:bedrock:us-west-2::foundation-model/anthropic.claude-3-5-sonnet-20241022-v2:0',
+        'arn:aws:bedrock:us-east-1:275631959608:inference-profile/us.anthropic.claude-3-5-sonnet-20241022-v2:0'
```

Comment:
Hardcoded AWS account id, pls replace with a dynamic value (`cdk.Aws.ACCOUNT_ID`) or wildcard and use least privilege principles.
If the region must be us-east-1 only, please add this as a limitation in the README (see comment above). Otherwise use dynamic value